sound/soc/sof/ipc.c

```diff
@@ -244,14 +249,18 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 {
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct snd_sof_ipc_msg *msg;
+	int ret;
+
+	mutex_lock(&ipc->tx_mutex);
 
 	spin_lock_irq(&sdev->ipc_lock);
 
 	/* get an empty message */
 	msg = msg_get_empty(ipc);
 	if (!msg) {
 		spin_unlock_irq(&sdev->ipc_lock);
-		return -EBUSY;
+		ret = -EBUSY;
```

Comment:
didn't we say BUSY wasn't the right return value? I am confused maybe

Reply:
No, I think it was a comment for an EAGAIN error code. This specific hunk isn't adding or changing the return code, it's just replacing a "return" with a "goto" to have a chance to unlock the mutex. A later patch removes this error code completely, once lists of messages are removed too.